### PR TITLE
feat(in-app-notification): Create NotificationSender to manage notification providers

### LINKED ISSUE #9245: In-App Error Notifications - Create NotificationSender
As we are going to have at least two different notification providers (system and in-app), it is important to have a manager to simplify the notification triggering.

The `NotificationSender` will be responsible for receiving a notification object and triggering the notification using the appropriate provider.

### PR DESCRIPTION
**Pull request by rafaeltonholo**
Resolves #9245 

This pull request introduces a `NotificationSender` responsible for managing and dispatching notifications through various providers. As the application now supports both in-app and system notifications, this manager simplifies the process of triggering a notification by abstracting the provider-specific logic.

The `NotificationSender` receives a notification object and determines the appropriate provider to use for delivery.

### Changes Included

- `NotificationSender` Implementation: Core manager for handling notification logic.
- `InAppNotificationNotifier`: Implementation for displaying notifications within the app UI.
- `SystemNotificationNotifier`: Implementation for triggering native system notifications.
- Enhanced Notification Features:
    - Added support for notification actions, allowing for interactive notifications.
    - Included notification icons and custom styling capabilities.
- Debug Tools: A new secret debug settings screen has been added to facilitate testing of the notification system and other options in the future.
- UI Components:
    - Added a horizontal divider atom to the design system.
    - Resolved rendering issues with theme previews.

### CODE REVIEW

File: core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/divider/HorizontalDivider.kt [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.unit.Dp
+import app.k9mail.core.ui.compose.theme2.MainTheme
+import androidx.compose.material3.HorizontalDivider as MaterialHorizontalDivider
+
+@Composable
+fun HorizontalDivider(
```
Comment (wmontwe) [inline core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/atom/divider/HorizontalDivider.kt:11]: There is already a `DividerHorizontal`

File: core/ui/compose/theme2/common/src/main/kotlin/app/k9mail/core/ui/compose/theme2/ThemeSizes.kt [hunk 2]
```diff
@@ -6,6 +6,7 @@ import androidx.compose.ui.unit.Dp
 
 @Immutable
 data class ThemeSizes(
+    val tiny: Dp,
```
Comment (wmontwe) [inline core/ui/compose/theme2/common/src/main/kotlin/app/k9mail/core/ui/compose/theme2/ThemeSizes.kt:9]: I think this should be called `line` or `divider`.

File: feature/debug-settings/build.gradle.kts [hunk 3]
```diff
@@ -0,0 +1,19 @@
+plugins {
+    id(ThunderbirdPlugins.Library.androidCompose)
+}
+
+android {
+    namespace = "net.thunderbird.feature.debugSettings"
```
Comment (wmontwe) [inline feature/debug-settings/build.gradle.kts:6]: This introduces camelCase package names to the project, I think we should stick to the existing pattern and split `debugSettings` into `debug.settings` for consistency.

File: feature/debug-settings/src/debug/kotlin/net/thunderbird/feature/debugSettings/DebugSectionPreview.kt [hunk 4]
```diff
@@ -0,0 +1,24 @@
+package net.thunderbird.feature.debugSettings
```
Comment (wmontwe) [inline feature/debug-settings/src/debug/kotlin/net/thunderbird/feature/debugSettings/DebugSectionPreview.kt:1]: This introduces camelCase package names to the project, I think we should stick to the existing pattern and split `debugSettings` into `debug.settings` for consistency.

File: feature/debug-settings/src/debug/kotlin/net/thunderbird/feature/debugSettings/inject/FeatureDebugSettingsModule.kt [hunk 5]
```diff
@@ -0,0 +1,17 @@
+package net.thunderbird.feature.debugSettings.inject
```
Comment (wmontwe) [inline feature/debug-settings/src/debug/kotlin/net/thunderbird/feature/debugSettings/inject/FeatureDebugSettingsModule.kt:1]: We should use a common package name for injection declarations. At the moment there is either none or `di` in use.

File: feature/debug-settings/src/main/kotlin/net/thunderbird/feature/debugSettings/notification/DebugNotificationSection.kt [hunk 6]
```diff
@@ -0,0 +1,225 @@
...
+import net.thunderbird.feature.debugSettings.notification.DebugNotificationSectionContract.ViewModel
+import net.thunderbird.feature.mail.account.api.BaseAccount
+import net.thunderbird.feature.notification.api.content.MailNotification
+import net.thunderbird.feature.notification.api.content.Notification
+import org.koin.androidx.compose.koinViewModel
+
+private const val UUID_MAX_CHAR_DISPLAY = 4
+
+@Composable
+internal fun DebugNotificationSection(
```
Comment (wmontwe) [inline feature/debug-settings/src/main/kotlin/net/thunderbird/feature/debugSettings/notification/DebugNotificationSection.kt:32]: Nice 👍

File: core/common/build.gradle.kts [hunk 7]
```diff
@@ -1,5 +1,6 @@
 plugins {
     id(ThunderbirdPlugins.Library.kmp)
+    id("kotlin-parcelize")
```
Comment (wmontwe) [inline core/common/build.gradle.kts:3]: This should be `alias(libs.plugins.kotlin.parcelize)`

File: feature/notification/api/build.gradle.kts [hunk 8]
```diff
@@ -1,5 +1,8 @@
+import org.jetbrains.kotlin.gradle.internal.config.LanguageFeature
+
 plugins {
     id(ThunderbirdPlugins.Library.kmpCompose)
+    id("kotlin-parcelize")
```
Comment (wmontwe) [inline feature/notification/api/build.gradle.kts:5]: This should be `alias(libs.plugins.kotlin.parcelize)`

File: core/common/build.gradle.kts [hunk 9]
```diff
@@ -28,4 +29,12 @@ kotlin {
...
         )
     }
+
+    compilerOptions {
+        freeCompilerArgs.addAll(
+            "-P",
+            "plugin:org.jetbrains.kotlin.parcelize:additionalAnnotation=net.thunderbird.core.common.io.KmpParcelize",
+            "-Xexpect-actual-classes",
+        )
+    }
```
Comment (wmontwe) [inline core/common/build.gradle.kts:39]: The compiler section is now doubled, see lines 25-31

File: feature/notification/api/build.gradle.kts [hunk 10]
```diff
@@ -8,6 +11,20 @@ kotlin {
             implementation(projects.core.common)
             implementation(projects.core.outcome)
         }
+        androidMain.dependencies {
+            implementation(projects.core.ui.compose.designsystem)
+        }
+    }
```
Comment (wmontwe) [inline feature/notification/api/build.gradle.kts:17]: What about adding the icons here directly instead of the design system? Otherwise we could consider moving icons to their own module.

File: feature/notification/api/src/commonMain/composeResources/values/strings.xml [hunk 11]
```diff
@@ -1,6 +1,6 @@
 <?xml version="1.0" encoding="UTF-8"?>
 <resources>
-    <!-- TODO: Perform a cleanup removing the other resources from the other modules with the same name once the notification system is consolidated. -->
+    <!-- TODO(#9418): Perform a cleanup removing the other resources from the other modules with the same name once the notification system is consolidated. -->
```
Comment (wmontwe) [inline feature/notification/api/src/commonMain/composeResources/values/strings.xml:3]: 👍

File: legacy/ui/legacy/build.gradle.kts [hunk 12]
```diff
@@ -19,6 +19,7 @@ dependencies {
     implementation(projects.core.ui.theme.api)
     implementation(projects.feature.launcher)
     implementation(projects.core.common)
+    implementation(projects.core.outcome)
```
Comment (wmontwe) [inline legacy/ui/legacy/build.gradle.kts:22]: It this required?

**Review by wmontwe — COMMENTED** [review]
Thank you for your work on the notification system overhaul. I've reviewed the code and have several observations that I'd like to share to help improve both this PR and future contributions.

I'm wondering if the debug section should work as it doesn't trigger any notifications on my side.

## Complexity and API Surface

The notification system you've designed is quite comprehensive, but the API surface is unnecessarily large and complex. For example:

- The hierarchy of notification types (`SystemNotification`, `InAppNotification`, various subtypes like `MailNotification.NewMail.SingleMail`, `PushServiceNotification`, etc.) creates a complex inheritance tree
- The extensive use of sealed interfaces and classes adds layers of abstraction that make the code harder to follow
- The notification styling system with multiple styles (`NotificationStyle.System.BigTextStyle`, `NotificationStyle.System.InboxStyle`, etc.) adds another dimension of complexity

Consider simplifying the API by:
- Reducing the number of notification types and consolidating similar ones
- Flattening the inheritance hierarchy where possible
- Using composition over inheritance for specialized behavior

## Test Coverage

Despite the complexity of the implementation, the tests are sparse. This makes it difficult to verify the behavior of the code changes and increases the risk of regressions. For a system as critical as notifications, comprehensive test coverage is essential.

I would recommend:
- Adding unit tests for each notification type
- Creating integration tests that verify the end-to-end notification flow
- Adding tests for edge cases (e.g., permission denied, notification channels not available)
- Including tests for the command pattern implementation

## PR Size and Structure

The PR is too large and tackles too many concerns at once. This makes it difficult to review thoroughly and increases the chance of issues slipping through. Breaking it down into smaller, focused PRs would have been more manageable.

Specific components that could have been separate PRs:

1. **Debug Tools**: The `feature/debug-settings` module with its comprehensive UI for testing notifications could have been a standalone PR
2. **Parcelize Support**: The KMP parcelize implementation in `core/common` could have been separated
3. **Notification API**: The base interfaces and types could have been introduced first
4. **System Notification Implementation**: Android-specific implementation
5. **In-App Notification Implementation**: The broadcast mechanism
6. **Animation Resources**: The animated notification icons
7. **Legacy Integration**: Connection with existing notification settings

## Architectural Concerns

The command pattern implementation (`NotificationCommandFactory`, `SystemNotificationCommand`, etc.) is a good approach for handling different notification types, but it adds complexity without clear documentation on how it fits into the larger architecture.

## Recommendations

For this PR:
1. Consider simplifying the API before merging
2. Add more tests to cover critical functionality
3. Provide better documentation explaining the architectural decisions

For future PRs:
1. Break down large features into smaller, focused PRs
2. Implement core functionality before adding specialized features
3. Add tests alongside implementation
4. Consider using feature flags to merge incomplete features safely

I appreciate the effort put into this comprehensive notification system, but addressing these concerns would make it more maintainable and easier to integrate into the codebase.

**Comment by rafaeltonholo** [comment]
Hi @wmontwe, thanks for reviewing the PR! I will answer your comments/concerns in separate comments, so it makes it easier to discuss each section

**Comment by rafaeltonholo** [comment]
### PR Size and Structure
I agree with you, the PR is quite long. Would you prefer me splitting it into different PRs right now so it would be easier to focus on some areas?

**Comment by rafaeltonholo** [comment]
## Complexity and API Surface
### About the hierarchy: 
The concept of using sealed classes with a nested hierarchy for notifications is designed to make it easier for developers to locate a specific notification based on its group or name.

For instance, if someone wants to trigger a notification after a new email is sent but encounters a failure, they could search for:

- `MailNotification`
- `SendFail`

By searching for `MailNotification`, they would also find the relevant nested notification, `SendFail`, making it straightforward to send the appropriate notification.

I find this approach to be organized rather than complex. However, if you believe that flattening the notifications and creating a Notification class without utilizing sealed classes would enhance the API surface, I can certainly do that.

### About the Notification styling system:
I need to highlight that there is a difference between a Notification type (`AuthenticationErrorNotification`) and a Notification style (`NotificationStyle.System.BigTextStyle`).

The same notification can be displayed via different providers (System and/or In-App) and with different styles. Additionally, in the case of an in-app notification, we could show the same notification using multiple styles.

The changes present in this PR only consider the system notification styles we are already using in the app. Nonetheless, this new API is making it a bit more platform-agnostic. Even though we use the same names as the Android Notification system, we could implement the visual representation of it differently depending on the platform.

### Regarding the number of notification types and consolidating similar ones

The current types of notifications are a direct one-to-one mapping of each notification that we are already triggering in the app.

Apart from the `SingleMail` and `SummaryMail`, which I believe could be merged into a single notification, I don’t see many other types that could be combined. This is unless we decide to eliminate some notifications that we are currently sending to users.

I’m open to any suggestions you might have.

### Using composition over inheritance for specialized behaviour
To be honest, I'm not sure where that would be applied here. Could you please point out where you believe that could be applied?

**Comment by rafaeltonholo** [comment]
### Test Coverage
> Adding unit tests for each notification type

I will add them.

> Creating integration tests that verify the end-to-end notification flow

Sure, I'll try to add them.

> Adding tests for edge cases (e.g., permission denied, notification channels not available)

These are all related to the System Notification command execution and is going to be handled in a different task (#9391)

> Including tests for the command pattern implementation

I will add them.

**Comment by rafaeltonholo** [comment]
### Architectural Concerns
> adds complexity without clear documentation on how it fits into the larger architecture.

There is a task (#9252) that focuses on documenting the new API. Is that what you are referring to? 

Additionally, I would appreciate it if you could provide a few specific examples of what you would like me to emphasize in the documentation. Thank you!

**Comment by rafaeltonholo** [comment]
This PR will be split into smaller PRs to improve the review
